cmd/trails/commands/root.go

```
@@ -37,37 +43,42 @@ var rootCmd = &cobra.Command{
+		engineConfig := engine.DefaultConfig()
+		eng, err := engine.New(engineConfig, commandChan, eventChan, sessionManager, stateManager, containerManager, metrics, logger)
 		if err != nil {
-			return fmt.Errorf("failed to create session manager: %w", err)
+			return fmt.Errorf("failed to create engine: %w", err)
+		}
+
+		if err := eng.Start(ctx); err != nil {
+			return fmt.Errorf("failed to start engine: %w", err)
 		}
-		defer closeManager()
+		defer eng.Stop()
```

golangci-lint (1.64.8)

Error return value of `eng.Stop` is not checked

(errcheck)